okta/resource_okta_app_oauth.go

```diff
@@ -770,21 +770,31 @@ func buildAppOAuth(d *schema.ResourceData) *sdk.OpenIdConnectApplication {
 	}
 
 	refresh := &sdk.OpenIdConnectApplicationSettingsRefreshToken{}
-	hasRefresh := false
+	var hasRefresh bool
+	for _, grant_type := range grantTypes {
+		if grant_type == refreshToken {
+			hasRefresh = true
+		}
+	}
 	if rotate, ok := d.GetOk("refresh_token_rotation"); ok {
 		refresh.RotationType = rotate.(string)
-		hasRefresh = true
 	}
 
 	if leeway, ok := d.GetOk("refresh_token_leeway"); ok {
 		refresh.LeewayPtr = int64Ptr(leeway.(int))
-		hasRefresh = true
 	}
 
 	if hasRefresh {
 		app.Settings.OauthClient.RefreshToken = refresh
 	}
 
+	// TODU
```

Comment:
Should we add an error or a warning here?

Reply:
if it should have have a refresh value, but does not, that is an error is it not?